Gems/ROS2/Code/Source/Spawner/ROS2SpawnerComponent.cpp

```diff
@@ -110,12 +111,22 @@ namespace ROS2
     void ROS2SpawnerComponent::SpawnEntity(
         const SpawnEntityServiceHandle service_handle, const std::shared_ptr<rmw_request_id_t> header, const SpawnEntityRequest request)
     {
+        AZStd::string referenceFrame(request->reference_frame.c_str());
```

Comment:
Since the "wgs84" frame doest not exist in ROS (it is a hack to re-use gazebo spawn message), I would consider adding an option to disable this behavior. This option can be enabled by default since it is a cool feature!